# CAR-SALES dataset

Dữ liệu `car-sales.csv`: bộ dữ liệu chứa thông tin các ôtô cũ ở một cửa hàng bán ô tô cũ. 

Bạn hãy thực hiện các yêu cầu sau:


### 1. Đọc dữ liệu và cho biết dữ liệu có bao nhiêu dòng, bao nhiêu cột

Đáp án: (10 dòng, 5 cột)

In [49]:
# Your code here
import pandas as pd
import numpy as np
car = pd.read_csv('data/car-sales.csv', header=0)
car.head(10)

Manufacturer Colour  Odometer  Doors    Price
0       Toyota  White    150043      4   4000.0
1        Honda    Red     87899      4   5000.0
2       Toyota   Blue     32549      3   7000.0
3          BMW  Black     11179      5  22000.0
4       Nissan  White    213095      4   3500.0
5       Toyota  Green     99213      4   4500.0
6        Honda   Blue     45698      4   7500.0
7        Honda   Blue     54738      4   7000.0
8       Toyota  White     60000      4   6250.0
9       Nissan  White     31600      4   9700.0

### 2. Cho biết thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng

In [8]:
# Your code here
car.info()
# thuộc tính định tính: Manufacturer, Colour
# thuộc tính định lượng: Odometer, Doors, Price

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Manufacturer  10 non-null     object 
 1   Colour        10 non-null     object 
 2   Odometer      10 non-null     int64  
 3   Doors         10 non-null     int64  
 4   Price         10 non-null     float64
dtypes: float64(1), int64(2), object(2)
memory usage: 528.0+ bytes


### 3. Trong dữ liệu có thông tin của bao nhiêu xe?

Đáp án: 10 xe

In [12]:
# Your code here
car.count()

Manufacturer    10
Colour          10
Odometer        10
Doors           10
Price           10
dtype: int64

### 4. Đếm số xe theo theo hãng

Đáp án: Toyota: 10 xe, Honda: 3 xe, Nissan: 2 xe, BMW: 1 xe

In [13]:
# Your code here
car['Manufacturer'].value_counts()

Toyota    4
Honda     3
Nissan    2
BMW       1
Name: Manufacturer, dtype: int64

### 5. Đếm số xe theo màu


In [14]:
# Your code here
car['Colour'].value_counts()

White    4
Blue     3
Black    1
Green    1
Red      1
Name: Colour, dtype: int64

### 6. Tổng giá trị các xe trong cửa hàng

Đáp án: 76450 USD


In [22]:
# Your code here
car['Price'].sum()

76450.0

### 7. Top 3 xe có giá cao nhất


In [21]:
# Your code here
car.sort_values(by = 'Price', ascending=False).head(3)

Manufacturer Colour  Odometer  Doors    Price
3          BMW  Black     11179      5  22000.0
9       Nissan  White     31600      4   9700.0
6        Honda   Blue     45698      4   7500.0

### 8. Top 3 xe đã chạy nhiều cây số (km)nhất (Odometer)

Đáp án:
- Xe Nissan, White, 213095 km
- Xe Toyota, White, 150043 km
- Xe Toyota, Green, 99213 km

In [25]:
# Your code here
car = car.sort_values(by = 'Odometer', ascending=False)
car[['Manufacturer', 'Odometer']].head()

Manufacturer  Odometer
4       Nissan    213095
0       Toyota    150043
5       Toyota     99213

### 9. Khảo sát xe của hãng Honda

- Tính tỷ lệ xe Honda trong dữ liệu của cửa hàng (0.3)
- Cho biết giá trung bình của xe hãng Honda của cửa hàng (6500)
- Cho biết các xe của hãng Honda trong cửa hàng có những màu sắc nào (Red, Blue)

In [56]:
# Your code here
a = car[car.Manufacturer == 'Honda']
print(len(a)/len(car))
print(a['Price'].mean())
c = car

0.3
6500.0
10


# 10. Tính toán các giá trị sau của  thuộc tính giá ('Price')

- Giá trung bình các xe (7645)
- Giá trị trung vị các xe (6625)
- Miền giá trị của xe (18500)
- Miền phân vị của xe (2750)
- Điểm phân vị 0.4 và 0.6 của giá ([5750, 7000])

In [29]:
car['Price'].describe()

count       3.0
mean     4000.0
std       500.0
min      3500.0
25%      3750.0
50%      4000.0
75%      4250.0
max      4500.0
Name: Price, dtype: float64

---

# TIPS dataset

Dữ liệu chứa trong file `tips.csv` chứa thông tin tiền tip của khách hàng tại một nhà hàng. Dữ liệu có các thông tin sau:

- `total_bill`: tổng hoá đơn (USD)
- `tip`: tiền tip/ tiền boa (USD)
- `sex`: giới tính người thanh toán hóa đơn
- `smoker`: có hút thuốc hay không
- `day`: ngày trong tuần
- `time`: thời gian dùng bữa(buổi sáng, buổi trưa, buổi tối)
- `size`: số lượng người tham gia

Bạn hãy thực hiện thống kê mô tả để tìm hiểu về hành vi tip của khách hàng tại nhà hàng

1. Đọc dữ liệu


In [4]:
# Your code here
import pandas as pd
import numpy as np
tip = pd.read_csv('data/tips.csv', header=0)
tip.head(10)

Unnamed: 0  total_bill   tip     sex smoker  day    time  size
0           0       16.99  1.01  Female     No  Sun  Dinner     2
1           1       10.34  1.66    Male     No  Sun  Dinner     3
2           2       21.01  3.50    Male     No  Sun  Dinner     3
3           3       23.68  3.31    Male     No  Sun  Dinner     2
4           4       24.59  3.61  Female     No  Sun  Dinner     4
5           5       25.29  4.71    Male     No  Sun  Dinner     4
6           6        8.77  2.00    Male     No  Sun  Dinner     2
7           7       26.88  3.12    Male     No  Sun  Dinner     4
8           8       15.04  1.96    Male     No  Sun  Dinner     2
9           9       14.78  3.23    Male     No  Sun  Dinner     2

2. Xoá cột `Unnamed: 0`


In [81]:
# Your code here
tip = tip.drop('Unnamed: 0', axis = 1)
tip.head(10)

total_bill   tip     sex smoker  day    time  size
0       16.99  1.01  Female     No  Sun  Dinner     2
1       10.34  1.66    Male     No  Sun  Dinner     3
2       21.01  3.50    Male     No  Sun  Dinner     3
3       23.68  3.31    Male     No  Sun  Dinner     2
4       24.59  3.61  Female     No  Sun  Dinner     4
5       25.29  4.71    Male     No  Sun  Dinner     4
6        8.77  2.00    Male     No  Sun  Dinner     2
7       26.88  3.12    Male     No  Sun  Dinner     4
8       15.04  1.96    Male     No  Sun  Dinner     2
9       14.78  3.23    Male     No  Sun  Dinner     2

3. Cho biết dữ liệu có bao nhiêu dòng, bao nhiêu thuộc tính.


In [10]:
# Your code here
tip.shape

(244, 7)

4. Trong các thuộc tính, thuộc tính nào là thuộc tính phân loại, thuộc tính nào là thuộc tính định lượng


In [11]:
# Your code here
tip.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244 entries, 0 to 243
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   total_bill  244 non-null    float64
 1   tip         244 non-null    float64
 2   sex         244 non-null    object 
 3   smoker      244 non-null    object 
 4   day         244 non-null    object 
 5   time        244 non-null    object 
 6   size        244 non-null    int64  
dtypes: float64(2), int64(1), object(4)
memory usage: 13.5+ KB


5. Có bao nhiêu nam, bao nhiêu nữ. Tỷ lệ bao nhiêu?


In [46]:
# Your code here
female = tip['sex'].value_counts()['Female']
male = tip['sex'].value_counts()['Male']
countt = tip['sex'].count()
print("Số nam: ", male)
print("Tỉ lệ: ", round(male *100 / countt, 2), "%")
print("Số nữ: ", female)
print("Tỉ lệ: ", round(female *100 / countt, 2), "%")

Số nam:  157
Tỉ lệ:  64.34 %
Số nữ:  87
Tỉ lệ:  35.66 %


6. Có bao nhiêu người hút thuốc. Tỷ lệ bao nhiêu?


In [48]:
# Your code here
yes = tip['smoker'].value_counts()['Yes']
countt = tip['smoker'].count()
print("Số người hút thuốc: ", yes)
print("Tỉ lệ: ", round(yes*100 / summ, 2), "%")

Số người hút thuốc:  93
Tỉ lệ:  38.11 %


7. Ngày nào trong tuần khách hàng hay đến dùng bữa nhất. Tỷ lệ bao nhiêu?

In [70]:
# Your code here
tip_day = tip.groupby("day").size()
tip_day = tip_day.sort_values(ascending=False).head(1)
maxx = tip_day.max()
summ = tip['day'].count()
print("Ngày khách hàng hay đến dùng bữa nhất: ")
print(tip_day)
print("Tỉ lệ: ", round(maxx*100/summ), "%")

Ngày khách hàng hay đến dùng bữa nhất: 
day
Sat    87
dtype: int64
Tỉ lệ:  36 %


8. Nhà hàng phục vụ bữa ăn nào nhiều nhất (Breakfast, Lunch, Dinner)?

In [84]:
# Your code here
tip_time = tip.groupby("time").size()
tip_time = tip_time.sort_values(ascending=False).head(1)
print("Nhà hàng phục vụ bữa ăn nhiều nhất: ")
print(tip_time)

Nhà hàng phục vụ bữa ăn nhiều nhất: 
time
Dinner    176
dtype: int64


9. Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất là bao nhiêu người?


In [96]:
# Your code here
tip_size = tip.groupby("size").size()
tip_size= tip_size.sort_values(ascending=False).head(1)
print("Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất: ")
print(tip_size)

Nhóm khách hàng dùng bữa tại nhà hàng nhiều nhất: 
size
2    156
dtype: int64


10. Trung bình trong một bữa ăn nhà hàng phục vụ mấy khách?

In [110]:
# Your code here
sizee = tip['size'].sum()
sizeee = tip['time'].count()
print(sizee / sizeee)

2.569672131147541


---